Course/Course/Presentation/Video/PlayerViewController.swift

```diff
@@ -52,40 +49,66 @@ struct PlayerViewController: UIViewControllerRepresentable {
         return controller
     }
     
-    private func addPeriodicTimeObserver(
-        _ controller: AVPlayerViewController,
-        currentProgress: @escaping ((Float, Double) -> Void)
-    ) {
-        let interval = CMTime(
-            seconds: 0.1,
-            preferredTimescale: CMTimeScale(NSEC_PER_SEC)
-        )
-        
-        self.controller.player?.addPeriodicTimeObserver(forInterval: interval, queue: .main) { time in
-            var progress: Float = .zero
-            let currentSeconds = CMTimeGetSeconds(time)
-            guard let duration = controller.player?.currentItem?.duration else { return }
-            let totalSeconds = CMTimeGetSeconds(duration)
-            progress = Float(currentSeconds / totalSeconds)
-            currentProgress(progress, currentSeconds)
-        }
-    }
-    
     func updateUIViewController(_ playerController: AVPlayerViewController, context: Context) {
         DispatchQueue.main.async {
```

Comment:
@forgotvas As we are here, how about defining `updateUIViewController` as `@MainActor`?

Reply:
@saeedbashir to be honestly i don't think that `DispatchQueue.main.async` required here, because updateUIViewController it's a swiftUI view life cycle method that should guarantee that it will be called from main thread. I will try to test that without dispatch and will commit changes.

Reply:
@saeedbashir have removed `DispatchQueue.main.async` and tested with enabled `Main thread checker` (enabled by default) and `Thread sanitizer` diagnostics tools - no problem there.